pkg/service/v1alpha1/node.go

```diff
@@ -119,29 +120,43 @@ verifyPublish:
 		// which implies that either the mount operation is complete
 		// or under progress.
 		// Lets verify if the mount is already completed
-		if info.Spec.Volume.DevicePath != "" {
+		if info.Spec.Volume.MountPath != mountPath {
+			utils.VolumesListLock.Unlock()
+			if !reVerified {
+				time.Sleep(
```

Comment:
I see that code here waits and does the same checks.. are there other go routines that can changes these conditions while code waits here? some comment would help if so

Reply:
Yes, we are waiting for other goroutines to update this condition.
Comment has been added.